### DISCUSSION TITLE
Trying to understand activity, impact, component

### DISCUSSION THREAD

**Discussion by psascue** [aws-solutions-library-samples/guidance-for-aws-sustainability-insights-framework#18] [Q&A] [answered]
Hi, we are trying to understand how the activity, impact, component objects work within SIF. We have an idea of their hierarchy though the examples. Now, we are trying to add our own but we have a couple of questions:

In [this example for natural gas](https://github.com/aws-solutions-library-samples/guidance-for-aws-sustainability-insights-framework/blob/main/docs/walkthrough.md#2--emission-factors), it has 2 impacts with 3 components each.

- where are the component values derived from? are they standards, or are they pre-calculated? if so, how? (we found [this published EF inventory by EPA](https://www.epa.gov/system/files/documents/2023-04/emission-factors_sept2021.pdf) but the values for natural gas doesn't match with the example)
- For context, we want to try adding a pipeline that takes power usage and outputs equivalent GHG emissions. If our understanding is correct, we need to identify the emission factors published by either the power plant (if available) or the geographic location and add it as an activity?

**Comment by boardthatpowder** [comment, marked as answer]
Hi @psascue 

Samples provided in the walkthrough may not reflect real world published standards. They are there to provide guidance on how to use the API's.

Note that SIF is designed to be agnostic to the GHG emissions standard that you intend to use, and does not prefer any specific framework over another. 

The [published EF inventory by EPA](https://www.epa.gov/system/files/documents/2023-04/emission-factors_sept2021.pdf) document you referenced is a good example of formal published emissions factors you could use. If you want to go a level deeper with that particular standard there is an [accompanying publication](https://www.epa.gov/sites/default/files/2020-12/documents/stationaryemissions.pdf) that describes how to use those emission factors within calculations. 

What you need to understand for the framework you intend to use is how to perform the calculation, what emission factors you need, and what additional datasets may be required. Continuing on from the [walkthrough](https://github.com/aws-solutions-library-samples/guidance-for-aws-sustainability-insights-framework/blob/main/docs/walkthrough.md#2--emission-factors) we provide, that gives an example of identifying and creating the [emission factors](https://github.com/aws-solutions-library-samples/guidance-for-aws-sustainability-insights-framework/blob/main/docs/walkthrough.md#2--emission-factors), identifying and creating additional [reference datasets](https://github.com/aws-solutions-library-samples/guidance-for-aws-sustainability-insights-framework/blob/main/docs/walkthrough.md#3-reference-datasets) (e.g. the GWP), then identifying and defining the [calculations](https://github.com/aws-solutions-library-samples/guidance-for-aws-sustainability-insights-framework/blob/main/docs/walkthrough.md#4-calculations). Once those are defined within the framework, you can create a [pipeline](https://github.com/aws-solutions-library-samples/guidance-for-aws-sustainability-insights-framework/blob/main/docs/walkthrough.md#5-pipelines) to describe how to transform the data, then submit data to be [processed](https://github.com/aws-solutions-library-samples/guidance-for-aws-sustainability-insights-framework/blob/main/docs/walkthrough.md#6-pipeline-executions-company-1).

**Reply by psascue** [reply]
hi @boardthatpowder , thanks for confirming! we initially got stuck trying to line up the values used in the examples to the published values to make sure we understand which values go where. 

Thanks for the guide, we can continue our review from here.
